diff --git a/meson.build b/meson.build
index 6851b1f7..eb535217 100644
--- a/meson.build
+++ b/meson.build
@@ -69,13 +69,9 @@ libui_project_link_args = []
 if libui_OS == 'darwin'
 	libui_darwin_langs = ['c', 'objc']
 
-	libui_macosx_version_min = '-mmacosx-version-min=10.8'
-	add_global_arguments(libui_macosx_version_min, language: libui_darwin_langs)
-	add_global_link_arguments(libui_macosx_version_min, language: libui_darwin_langs)
-
-	libui_arch = ['-arch', 'x86_64', '-arch', 'arm64']
-	add_global_arguments(libui_arch, language: libui_darwin_langs)
-	add_global_link_arguments(libui_arch, language: libui_darwin_langs)
+	libui_macosx_version_min = '-mmacosx-version-min=10.9'
+	add_project_arguments(libui_macosx_version_min, language: libui_darwin_langs)
+	add_project_link_arguments(libui_macosx_version_min, language: libui_darwin_langs)
 endif
 
 if libui_MSVC
